src/crimson/net/ProtocolV2.cc

```diff
@@ -1658,7 +1658,7 @@ void ProtocolV2::trigger_replacing(bool reconnect,
         reset_session(true);
       }
       protocol_timer.cancel();
-      return std::move(execution_done);
+      return execution_done.get_future();
```

Comment:
@cyx1231st could you put more details regarding 

- what the problem we are facing
- why s/future/shared_future/ is able to this issue

?

Reply:
It is to fix the assertion failure in seastar when certain failure happened:
```
[test_v2_lossless_reaccept_fault -- MESSAGE_READ, SESSION_RECONNECT_OK_WRITE]
...
unittest_seastar_messenger: /root/ceph/src/seastar/include/seastar/core/future.hh:284:
  void seastar::future_state_base::set_exception(std::__exception_ptr::exception_ptr&&):
    Assertion `_u.st == state::future' failed.
```
Added to the commit message.